### DISCUSSION TITLE
Can't read files in root directory in deployment

### DISCUSSION THREAD

**Discussion by nitlix** [vercel/community#2894] [Help]
### Summary

Hello there! I'm currently trying to build my own Express Framework which would basically be reading a file located in `./styles/exampleComponent.css`. The style folder is in the root folder. It would be read from one of the files located in the node modules. But, when that does happen, an error is thrown, despite the file being there: 

<img width="552" alt="image" src="https://github.com/vercel/community/assets/66027553/15ef13fd-00c8-4b59-8100-a4772f54782b">

I'm basically using an express app, which starts up a server from a file in node_modules, and imports it in api/index.js.
Is there any way to allow those reads or basically just fix something to let it access them? Many thanks!

### Steps to Reproduce

Try creating an express app with a custom node module, and making the module read something from the root directory.

**Comment by amyegan** [comment]
Hi @Nitlix! I'm not sure of the project structure your repo uses, but I know Express projects need to be set up a little differently from normal to get them working in a serverless environment. That's the most common cause of problems.

You can find some more information about using Express.js with Vercel here: https://vercel.com/guides/using-express-with-vercel

**Reply by nitlix** [reply]
Hey @amyegan! Here's the project structure (only the necessary bits) that I tried to copy to make sure everything works:
Root: 
    1: /api
      2: index.js (Which is called)
   1: /node_modules
      2: /module_name
          3: /src
               4: /util
                    5: getCSS.js
    1: /styles


 The getCSS() function inside getCSS.js is called to get the css file inside the *styles* folder, however, for some reason it can't get to it 🤷‍♂️

Looking at the error log, the folder should probably be there as it's the right structure, but it's not...
What should I do? Thanks

**Reply by amyegan** [reply]
Try adding a `/public` directory and moving `/styles` into it. That will make the static style assets available on the live site.

**Reply by nitlix** [reply]
Alright, @amyegan, so is there no other way to be able to "tell vercel to keep certain folders" or somehow be able to access them without putting them in a `/public` directory? 

I am building a library somewhat similar to Next.JS and before deploying the app, it reads different data from those folders in root and it's necessary.
